### PR TITLE
DOC: Merge FAQ and gotcha (rebase of GH13768)

### LINKED ISSUE #9809: DOC: merge FAQ / Caveats & Gotchas
xref #9802
- I think making it FAQ would most appropriate
- want to think about the orderings
- need a release note to indicate that these have been merged

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Rebase and clean-up of https://github.com/pandas-dev/pandas/pull/13768

closes #9809

### CODE REVIEW

**Comment by jreback** [comment]
lgtm!

**Comment by jreback** [comment]
thanks!

File: doc/source/gotchas.rst [hunk 1]
```diff
@@ -1,18 +1,92 @@
 .. currentmodule:: pandas
 .. _gotchas:
 
+********************************
+Frequently Asked Questions (FAQ)
```
Comment (shoyer) [inline doc/source/gotchas.rst]: I'm not sure it really makes sense to call this new page "FAQ" until we rephrase the section headers as questions.

**Review by shoyer — COMMENTED** [review]
@jreback maybe we can leave things open for review for slightly longer next time? :)

File: doc/source/advanced.rst [hunk 2]
```diff
@@ -844,3 +844,136 @@ Of course if you need integer based selection, then use ``iloc``
...
+Miscellaneous indexing FAQ
+--------------------------
+
+Integer indexing
+~~~~~~~~~~~~~~~~
+
+Label-based indexing with integer axis labels is a thorny topic. It has been
+discussed heavily on mailing lists and among various members of the scientific
+Python community. In pandas, our general viewpoint is that labels matter more
+than integer locations. Therefore, with an integer axis index *only*
```
Comment (jorisvandenbossche) [inline doc/source/advanced.rst]: Yes, you are right. Originally this was a part about `ix`, and then this made more sense (the ix was replaced by loc after the deprecation)

I didn't really review the content of these parts, only moved them (apart from the comment that were already made on the other PR), but good to do that now as well.

File: doc/source/gotchas.rst (hunk 1 above)
Comment (jorisvandenbossche) [inline doc/source/gotchas.rst]: Ah, sorry that is a leftover from the other PR. I changed everything from FAQ back to gotchas, apart from the header.

**Pull request closed without merging**